tidy3d/web/webapi.py

```diff
@@ -152,18 +167,25 @@ def get_info(task_id: TaskId) -> TaskInfo:
     return TaskInfo(**{"taskId": task.task_id, **task.dict()})
 
 
-def start(task_id: TaskId) -> None:
+def start(
+    task_id: TaskId,
+    solver_version: str = None,
+    worker_group: str = None,
+    protocol_version: str = __version__,
+) -> None:
     """Start running the simulation associated with task.
 
     Parameters
     ----------
 
     task_id : str
         Unique identifier of task on server.  Returned by :meth:`upload`.
-    solver_version : str
-        Supply or override a specific solver version to the task.
-    verbose : bool = True
-        If `True`, will print progressbars and status, otherwise, will run silently.
+    solver_version: str = None
+        target solver version.
+    worker_group: str = None
+        worker group
+    protocol_version: str = None
+        protocol version
```

Comment:
I am actually confused, here and in a number of other places it seems to say `protocol_version: str = None` in the docstring while `protocol_version: str = __version__` in the function definition. What I also can't follow is what logic ensures that protocol_version is passed if solver_version is None and vice versa?

Reply:
On the first point, this is just how it was handed off to me, but I agree maybe we should make it actually `=None` in the method signature and then later set it to `__version__` if it's None?

On your second point about the logic, I dont quite follow. Is one of protocol version or solver version required? what logic should we implement because right now I dont know if there is any.

Reply:
So we figured out passing a `solver_version` doesn't work unless you also pass `prorocol_version=None`. I was under the impression that you would handle this internally, but I think I misunderstood that. You probably meant that you just tried passing `protocol_version=None` in web.run and that worked. In the previous webapi version, this was handled internally.

https://github.com/flexcompute/tidy3d/blob/027b422b84deb79446a013da4e6ebbb8dc0a2bdd/tidy3d/web/webapi.py#L191-L195

Reply:
I do feel like it doesn't hurt to handle internally here too.

Reply:
or I guess what do we do if `solver_version` and `protocol_version` is supplied for example?

Reply:
That was the logic before yeah. It seems like if they are both supplied the webapi returns an error, I guess because it's not sure which one to use. We could fix this behavior but that goes through modifying the webapi, not our python code.

Reply:
ok I changed it to do the same logic from within the simulation task submission and now all protocol versions are none by default (but set to __version__ if solver version is None)

https://github.com/flexcompute/tidy3d/blob/55d29a3e063aa61ae5978098b593a4dff1566685/tidy3d/web/simulation_task.py#L350-L353